src/Clapi/NamespaceTracker.hs

```diff
@@ -54,53 +57,47 @@ disallowedMsg o m = MsgError (msgPath' m) txt
         "Method ", T.pack $ show $ msgMethod' m, " forbidden when acting as ",
         roleTxt o]
 
-toOm :: TrackedMessage i -> (Ownership, Message)
-toOm tm = (tmOwnership tm, tmMessage tm)
-
 updateOwnerships ::
-    forall m i u. (Monad m, Ord i, Ord u, Show i) => AddrWithUser i u -> [TrackedMessage i] -> StateT (Owners (AddrWithUser i u)) m ()
-updateOwnerships i = mapM_ $ handle . toOm
+    forall m i u. (Monad m, Ord i, Ord u, Show i) => AddrWithUser i u -> [RoutableMessage i] -> StateT (Owners (AddrWithUser i u)) m ()
+updateOwnerships i = mapM_ $ handle
   where
-    handle :: (Ownership, Message) -> StateT (Owners (AddrWithUser i u)) m ()
-    handle (Unclaimed, MsgAssignType path _) =
+    handle :: RoutableMessage i -> StateT (Owners (AddrWithUser i u)) m ()
+    handle (RoutableMessage (Right Owner) (MsgAssignType path _)) =
         when (isNamespace path) (modify $ \x -> Map.insert (namespace path) i x)
-    handle (ownership, MsgDelete path) =
-        when
-            (ownership /= Client && isNamespace path)
-            (modify (Map.delete $ namespace path))
+    handle (RoutableMessage (Right Owner) (MsgDelete path)) =
+        when (isNamespace path) (modify (Map.delete $ namespace path))
     handle _ = return ()
 
+-- FIXME: not filtering anything out so refactor accordingly
```

Comment:
Do you mean that it _should_ be filtering things out and isn't? Or that it's no longer filtering things out but is still written like it is?

Reply:
That it is no longer filtering things out but is still written like it is.